src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -469,28 +420,36 @@ public boolean isPrivate() throws IOException {
         return getRepository().isPrivate();
     }
 
-
-    private <T> T parse(String response, Class<T> clazz) throws IOException {
-        ObjectMapper mapper = new ObjectMapper();
-        return mapper.readValue(response, clazz);
-    }
-
-    private String getRequest(String path) throws IOException {
+    /**
+     *
+     * @param path relative path for bitbucket server api
+     * @param clazz the class used for deserialization
+     * @param <T> generic type
+     * @return the response object
+     * @throws IOException
+     */
+    private <T> T getResponse(String path, Class<T> clazz) throws IOException {
         GetMethod httpget = new GetMethod(this.baseURL + path);
         HttpClient client = getHttpClient(getMethodHost(httpget));
         try {
```

Comment:
I would recommend to use the CloseableResponse with try-with-resources (http://stackoverflow.com/questions/21574478/what-is-the-difference-between-closeablehttpclient-and-httpclient-in-apache-http), that way you correctly release the connection even in case of errors